src/new.spec.js

```diff
@@ -111,5 +112,36 @@ describe("withStore", () => {
 
       expect(getState().baz).toBe("baz");
     });
+
+    it("cannot access itself", () => {
+      const { getState } = makeTestInstance({
+        initialState: () => ({}),
+        computed: {
+          circularComputed: ({ circularComputed }) => {},
+        },
+      });
+
+      expect(() => {
+        return getState().circularComputed;
+      }).toThrowError(new CircularComputedError("circularComputed"));
+    });
+
+    it("receive read-only props", () => {
+      const props = { bar: 2 };
+      const { getState } = makeTestInstance(
+        {
+          initialState: () => ({}),
+          computed: {
+            baz: (_, props) => {
+              assert(isReadOnly(props));
+              return props.bar * 2;
+            },
+          },
+        },
+        props
+      );
+
+      expect(getState().baz).toBe(4);
+    });
```

Comment:
You can probably merge this with the first state like we did for the render function.